File: beecrowd/2166 - Square Root of 2.cpp
#include <iostream> //  Permite entrada e saida
#include <iomanip>  //  Permite manipular a entrada e saida

int main(){
    int  n;
    //  Entrada
    std::cin >> n;

    double x = 0.0;
    //  Loop padrão
    for (short i = 0; i < n; ++i)
        x = 1.0/(2.0 + x);

    x = 1.0 + x;
    //  Saida normal do C, fixed para usar um ponto fixo inves de notação cientifica,
    //  endl para forçar saida
    std::cout << std::fixed << std::setprecision(10) << x << std::endl;
    
    return 0;
}

File: beecrowd/2166 - Square Root of 2.rs
use std::io;

fn main(){
    let mut n = String::new();

    //  Entrada
    io::stdin().read_line(&mut n).unwrap();
    
    let n: u8 = n.trim().parse().unwrap();
    
    let mut x = 0.0;
    

    //  Loop padrão
    for _ in 0..n{
        x = 1.0 /(2.0 + x);
    }
    
    x = 1.0 + x;
    println!("{:.10}",x); 
}

